### PR TITLE
Cache Google Oauth2 certificate

### PR DESCRIPTION
**Pull request by seallard**
## Description
Instead of fetching the certificate on each received request, we can cache it. This should reduce the response time for requests against our API. Google only rotates the certificate once per day.

The current implementation does not use that two certificates are returned from Google.

### Added
- Caching of Google OAuth certificate

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/server/api.py [hunk 1]
```diff
@@ -25,12 +27,49 @@
...
 
+cache = TTLCache(maxsize=1, ttl=3600)
+
+
+def get_certificate_ttl(response_data):
+    expires_header = response_data.headers.get("Expires")
+    expires = datetime.strptime(expires_header, "%a, %d %b %Y %H:%M:%S %Z")
+    ttl = int((expires - datetime.utcnow().replace(tzinfo=timezone.utc)).total_seconds())
+
+    return ttl
```
Comment (Vince-janv) [inline cg/server/api.py:43]: Docstring and type hints
```suggestion
def get_certificate_ttl(response_data) -> int:
    """Returns the time-to-live of given response."""
    expires_header: str = response_data.headers.get("Expires")
    expires: datetime = datetime.strptime(expires_header, "%a, %d %b %Y %H:%M:%S %Z")
    ttl: int = int((expires - datetime.utcnow().replace(tzinfo=timezone.utc)).total_seconds())
    
    return ttl
```

**Review by Vince-janv — APPROVED** [review]
Very neat solution! So `TTLCache` removes the cert when the ttl expires? Just add docstrings and type hints and I'm happy 😊

**Comment by seallard** [comment]
> Very neat solution! So `TTLCache` removes the cert when the ttl expires? Just add docstrings and type hints and I'm happy 😊

Yes, but it apparently only uses the ttl specified when creating the cache so I need to make some changes 🙁

**Comment by seallard** [comment]
Tested in stage

**Pull request merged**